op-e2e/actions/interop/interop_fork_test.go

```diff
@@ -115,14 +110,21 @@ func TestInteropUpgrade(gt *testing.T) {
 		system.ActSyncSupernode(t)
 	}, dsl.WithMapChainAssertions(dsl.WithSafeAdvancesTo(2)))
 
-	// Note that finalized remains nil
+	// TODO(15863): The initial state of the finalized head differs between supervisor
+	// and sequencer. In an action test setup, the sequencer considers
+	// genesis head as finalized from the start. The supervisor however sets
+	// the finalized head to a nil value until it it set. Seems worth fixing.
+	// The following will fail.
+	// system.ActSyncSupernode(t, dsl.WithChains(c), dsl.WithFinalizedSignal(), dsl.WithLatestSignal(), dsl.WithRequireFinalizedAdvances())
+	// syncAsserter.RequireSupChainHeadsBySyncStatus()
 	for _, c := range chains {
 		syncAsserter := superchainSyncAsserter.ChainAsserters[c.ChainID]
-		dsl.RequireSupervisorChainHeads(t, svr, c, syncAsserter.PrevStatus.UnsafeL2.ID(), syncAsserter.PrevStatus.CrossUnsafeL2.ID(), syncAsserter.PrevStatus.LocalSafeL2.ID(), syncAsserter.PrevStatus.SafeL2.ID(), eth.BlockID{Hash: common.BytesToHash([]byte{}), Number: 0})
+		dsl.RequireSupervisorChainHeads(t, svr, c, syncAsserter.PrevStatus.UnsafeL2.ID(), syncAsserter.PrevStatus.CrossUnsafeL2.ID(), syncAsserter.PrevStatus.LocalSafeL2.ID(), syncAsserter.PrevStatus.SafeL2.ID(), eth.BlockID{})
 	}
 
 	// Verify proofs agree
-	assertProgramOutputMatchesDerivationForBlockTimestamp(gt, system, system.Actors.ChainA.Sequencer.L2Unsafe().Time)
+	// TODO(#16166): Fix non-genesis Interop activation proofs
+	// assertProgramOutputMatchesDerivationForBlockTimestamp(gt, system, system.Actors.ChainA.Sequencer.L2Safe().Time)
```

Comment:
This is actually the same failure case as above, supervisor is failing to return a super root that we're requesting:

```
        	            	cross-derived-to-source failed for chain 900200: query is before first entry LinkEntry(source: BlockSeal(hash:0xd7073450474d7087067af9543164d2bf421a088bac8dc70e5940101ee7f74511, number:3, time:1748568053), derived: BlockSeal(hash:0xb2ac632babb01bf236db4e88455268acfb51921a0e8baf800ddb0b91ec7774b0, number:2, time:1748568021), invalidated: false): skipped data
```